src/realmd/AuthSocket.cpp

```diff
@@ -1115,137 +1100,117 @@ bool AuthSocket::_HandleRealmList()
-        case 5464:                                          // 1.11.2
-        case 5875:                                          // 1.12.1
-        case 6005:                                          // 1.12.2
-        case 6141:                                          // 1.12.3
-        {
-            pkt << uint32(0);                               // unused value
-            pkt << uint8(sRealmList.size());
-
-            for(RealmList::RealmMap::const_iterator  i = sRealmList.begin(); i != sRealmList.end(); ++i)
-            {
-                uint8 AmountOfCharacters;
-
-                // No SQL injection. id of realm is controlled by the database.
-                QueryResult *result = LoginDatabase.PQuery( "SELECT `numchars` FROM `realmcharacters` WHERE `realmid` = '%d' AND `acctid`='%u'", i->second.m_ID, _accountId);
-                if( result )
-                {
-                    Field *fields = result->Fetch();
-                    AmountOfCharacters = fields[0].GetUInt8();
-                    delete result;
-                }
-                else
-                    AmountOfCharacters = 0;
-
-                bool ok_build = std::find(i->second.realmbuilds.begin(), i->second.realmbuilds.end(), _build) != i->second.realmbuilds.end();
-
-                RealmBuildInfo const* buildInfo = ok_build ? FindBuildInfo(_build) : nullptr;
-                if (!buildInfo)
-                    buildInfo = &i->second.realmBuildInfo;
-
-                RealmFlags realmflags = i->second.realmflags;
-
-                // 1.x clients not support explicitly REALM_FLAG_SPECIFYBUILD, so manually form similar name as show in more recent clients
-                std::string name = i->first;
-                if (realmflags & REALM_FLAG_SPECIFYBUILD)
-                {
-                    char buf[20];
-                    snprintf(buf, 20," (%u,%u,%u)", buildInfo->major_version, buildInfo->minor_version, buildInfo->bugfix_version);
-                    name += buf;
-                }
-
-                // Show offline state for unsupported client builds and locked realms (1.x clients not support locked state show)
-                if (!ok_build || (i->second.allowedSecurityLevel > GetSecurityOn(i->second.m_ID)))
-                    realmflags = RealmFlags(realmflags | REALM_FLAG_OFFLINE);
-
-                pkt << uint32(i->second.icon);              // realm type
-                pkt << uint8(realmflags);                   // realmflags
-                pkt << name;                                // name
-                pkt << i->second.address;                   // address
-                pkt << float(i->second.populationLevel);
-                pkt << uint8(AmountOfCharacters);
-                pkt << uint8(i->second.timezone);           // realm category
-                pkt << uint8(0x00);                         // unk, may be realm number/id?
-            }
+  if (_build < CLIENT_BUILD_2_0_0)
```

Comment:
1, The tabulation looks weird here, it should be 4 spaces.
2. You've removed the build switch case, does that mean realmd will now let you login with any client version? This is not desired behavior, it should give the wrong version error when trying to connect with vanilla clients that mangosd does not yet support, or will never support. So that means any version older than 1.6 currently, or non-final major patch version (like 1.12.0 instead of 1.12.1 for example).

Reply:
1. I will fix it.
2. It is need to patch handler work. We need to allow unsupported clients to login so they can get patches. If we block login, as before, client will never accept CMD_XFER_INITIATE. Patch code, in this implementation, looks for update a intermediate version (1.11.0 ou 1.11.1 to 1.11.2, a supported build version).

Reply:
Can I suggest that we use a make command like `-DPATCHABLE=1` or `-DCUSTOMCLIENT=1`. The default for this should be 0 ofc. This will then not pollute the `CLIENT_BUILD_X_XX_X`  definitions.

It still has to interact with the version check ofc - but you then can use the preprocessor defined in make.

This way both systems are separated from each other and its not a hack

Reply:
At this moment, my code only updates intermediate versions if mpq patch is available. If not available, it gives a invalid version error as expected. In future I want to implement code for custom builds and I will follow your instructions.